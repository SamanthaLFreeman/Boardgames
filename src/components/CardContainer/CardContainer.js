import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

export const CardContainer = () => {
  return (
    <section></section>
  )
}

const mapStateToProps = state => ({
  games: state.games
})

export default connect(mapStateToProps, null)(CardContainer);