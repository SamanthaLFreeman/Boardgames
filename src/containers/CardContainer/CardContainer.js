import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Card from '../../components/Card/Card'

export const CardContainer = ({games, toggleFavorite, favoriteCheck, favorites}) => {
  const allCards = games.map(game => <Card game={game} key={game.id} toggleFavorite={toggleFavorite} />);
  const favoriteCards = favorites.map(game => <Card game={game} key={game.id} toggleFavorite={toggleFavorite} />);
  const displayCards = !favoriteCheck ? allCards : favoriteCards;
  return (
    <section className="cards-container">
      {displayCards}
    </section>
  )
}

const mapStateToProps = state => ({
  games: state.games,
  favorites: state.favorites
})

export default connect(mapStateToProps, null)(CardContainer);