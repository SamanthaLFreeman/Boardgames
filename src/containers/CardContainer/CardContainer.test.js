import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from './CardContainer';

describe('CardContainer', () => {
  let wrapper;
  const mockGames = [
    {
      name: 'Catan',
      id: 42
    }
  ]
  const mockFavorites = [
    {
      name: 'Takenoko',
      id: 41
    }
  ]
  const mockOwned = [
    {
      name: 'Dead of Winter',
      id: 43
    }
  ]
  const toggleFavoriteMock = jest.fn();
  const toggleOwnedMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<CardContainer 
    games={mockGames}
    favorites={mockFavorites}
    ownedGames={mockOwned}
    toggleFavorite={toggleFavoriteMock}
    toggleOwned={toggleOwnedMock}
    type='games' />)
  });

  it('should match the snapshot with the data passed through with type games', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with the data passed through with type favorites', () => {
    let wrapper = shallow(<CardContainer
      games={mockGames}
      favorites={mockFavorites}
      ownedGames={mockOwned}
      toggleFavorite={toggleFavoriteMock}
      toggleOwned={toggleOwnedMock}
      type='favorites' />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with the data passed through with type ownedGames', () => {
    let wrapper = shallow(<CardContainer
      games={mockGames}
      favorites={mockFavorites}
      ownedGames={mockOwned}
      toggleFavorite={toggleFavoriteMock}
      toggleOwned={toggleOwnedMock}
      type='ownedGames' />)
    expect(wrapper).toMatchSnapshot();
  });
});