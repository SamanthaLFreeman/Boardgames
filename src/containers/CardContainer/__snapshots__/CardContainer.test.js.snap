// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with the data passed through with type favorites 1`] = `
<section
  className="cards-container"
>
  <Connect(Card)
    game={
      Object {
        "id": 41,
        "name": "Takenoko",
      }
    }
    key="41"
    toggleFavorite={[MockFunction]}
    toggleOwned={[MockFunction]}
  />
</section>
`;

exports[`CardContainer should match the snapshot with the data passed through with type games 1`] = `
<section
  className="cards-container"
>
  <Connect(Card)
    game={
      Object {
        "id": 42,
        "name": "Catan",
      }
    }
    key="42"
    toggleFavorite={[MockFunction]}
    toggleOwned={[MockFunction]}
  />
</section>
`;

exports[`CardContainer should match the snapshot with the data passed through with type ownedGames 1`] = `
<section
  className="cards-container"
>
  <Connect(Card)
    game={
      Object {
        "id": 43,
        "name": "Dead of Winter",
      }
    }
    key="43"
    toggleFavorite={[MockFunction]}
    toggleOwned={[MockFunction]}
  />
</section>
`;
